src/Solver/Z3Solver.cpp

```
@@ -277,7 +277,12 @@ z3::expr Z3OpVisitor::visitConstantInt(const ConstantInt& op) {
 
 z3::expr Z3OpVisitor::visitConstantFloat(const ConstantFloat& op) {
   // TODO: Reimplement this correctly
-  return ctx->fpa_val(op.value().convertToFloat());
+  auto expr = z3::expr(
```

Is the TODO still accurate?

---

Yup. The above just converts to a double. Any bigger FP types will get silently truncated when they go through this.